base/ltfssaxes.dtx

```diff
@@ -192,503 +248,3158 @@
+\DeclareFontSeriesChangeRule {bx}{el}   {elx}  {}
+\DeclareFontSeriesChangeRule {bx}{l}    {lx}   {}
+\DeclareFontSeriesChangeRule {bx}{sl}   {slx}  {}
+\DeclareFontSeriesChangeRule {bx}{sb}   {sbx}  {bx}
+\DeclareFontSeriesChangeRule {bx}{b}    {bx}   {}
+\DeclareFontSeriesChangeRule {bx}{eb}   {ebx}  {bx}
+\DeclareFontSeriesChangeRule {bx}{ub}   {ubx}  {bx}
+\DeclareFontSeriesChangeRule {bx}{uc}   {buc}  {}
+\DeclareFontSeriesChangeRule {bx}{ec}   {bec}  {}
+\DeclareFontSeriesChangeRule {bx}{c}    {bc}   {}
+\DeclareFontSeriesChangeRule {bx}{sc}   {bsc}  {}
+\DeclareFontSeriesChangeRule {bx}{sx}   {bsx}  {}
+\DeclareFontSeriesChangeRule {bx}{x}    {bx}   {}
+\DeclareFontSeriesChangeRule {bx}{ex}   {bex}  {}
+\DeclareFontSeriesChangeRule {bx}{ux}   {bux}  {}
+%    \end{macrocode}
+%
+%    \begin{macrocode}
+\DeclareFontSeriesChangeRule {bex}{ul}  {ulex} {}
+\DeclareFontSeriesChangeRule {bex}{el}  {elex} {}
+\DeclareFontSeriesChangeRule {bex}{l}   {lex}  {}
+\DeclareFontSeriesChangeRule {bex}{sl}  {slex} {}
+\DeclareFontSeriesChangeRule {bex}{sb}  {sbex} {bex}
+\DeclareFontSeriesChangeRule {bex}{b}   {bex}  {}
+\DeclareFontSeriesChangeRule {bex}{eb}  {ebex} {bex}
+\DeclareFontSeriesChangeRule {bex}{ub}  {ubex} {bex}
+\DeclareFontSeriesChangeRule {bex}{uc}  {buc}  {}
+\DeclareFontSeriesChangeRule {bex}{ec}  {bec}  {}
+\DeclareFontSeriesChangeRule {bex}{c}   {bc}   {}
+\DeclareFontSeriesChangeRule {bex}{sc}  {bsc}  {}
+\DeclareFontSeriesChangeRule {bex}{sx}  {bsx}  {}
+\DeclareFontSeriesChangeRule {bex}{x}   {bx}   {}
+\DeclareFontSeriesChangeRule {bex}{ex}  {bex}  {}
+\DeclareFontSeriesChangeRule {bex}{ux}  {bux}  {}
+%    \end{macrocode}
+%
+%    \begin{macrocode}
+\DeclareFontSeriesChangeRule {bux}{ul}  {ulux} {}
+\DeclareFontSeriesChangeRule {bux}{el}  {elux} {}
+\DeclareFontSeriesChangeRule {bux}{l}   {lux}  {}
+\DeclareFontSeriesChangeRule {bux}{sl}  {slux} {}
+\DeclareFontSeriesChangeRule {bux}{sb}  {sbux} {bux}
+\DeclareFontSeriesChangeRule {bux}{b}   {bux}  {}
+\DeclareFontSeriesChangeRule {bux}{eb}  {ebux} {bux}
+\DeclareFontSeriesChangeRule {bux}{ub}  {ubux} {bux}
+\DeclareFontSeriesChangeRule {bux}{uc}  {buc}  {}
+\DeclareFontSeriesChangeRule {bux}{ec}  {bec}  {}
+\DeclareFontSeriesChangeRule {bux}{c}   {bc}   {}
+\DeclareFontSeriesChangeRule {bux}{sc}  {bsc}  {}
+\DeclareFontSeriesChangeRule {bux}{sx}  {bsx}  {}
+\DeclareFontSeriesChangeRule {bux}{x}   {bx}   {}
+\DeclareFontSeriesChangeRule {bux}{ex}  {bex}  {}
+\DeclareFontSeriesChangeRule {bux}{ux}  {bux}  {}
+%    \end{macrocode}
+%
+%    \begin{macrocode}
+\DeclareFontSeriesChangeRule {ebuc}{ul} {uluc} {}
+\DeclareFontSeriesChangeRule {ebuc}{el} {eluc} {}
+\DeclareFontSeriesChangeRule {ebuc}{l}  {luc}  {}
+\DeclareFontSeriesChangeRule {ebuc}{sl} {sluc} {}
+\DeclareFontSeriesChangeRule {ebuc}{sb} {sbuc} {buc}
+\DeclareFontSeriesChangeRule {ebuc}{b}  {buc}  {}
+\DeclareFontSeriesChangeRule {ebuc}{eb} {ebuc} {buc}
+\DeclareFontSeriesChangeRule {ebuc}{ub} {ubuc} {buc}
```

Comment:
```suggestion
\DeclareFontSeriesChangeRule {ebuc}{ub} {ubuc} {ebuc}
```

I think it is better to use `eb` instead of `b` in the fourth argument: `eb` is a better approximation for `ub` and since the fourth argument is then identical to the first, we can assume that this font face actually exists.

Reply:
good argument

Reply:
I don't follow this. Why is it OK to assume the font series specified in the first argument actually exists? Just that it is the current series? 

In general, it is not safe to assume that the current font name reflects the actual properties of the current font. So, unless I'm missing something, there is no guarantee that the current series exists in anything but name.

Reply:
I don't follow this pattern of argument. There's no guarantee the current font name reflects the actual properties of the current font, so why assume the first argument represents a series which exists in anything but name?

Reply:
We are talking about a fallback behavior, i.e., it happens only if something is wrong.
The assumption made is that when the rule is applied the \f@series represents what the current font implements. -  - That is true, if  ebuc was reached by applying \fontseries{eb} weight and \fontseries{uc} width seperately, because if not  a substitution rule would have kicked in earlier. It is also true if it was reached via \fontseries{ebuc} directly and the font actually existed.
- It would be false, if  \fontseries{ebuc}  was used and  the font didn't exist.

Assuming it is false the the current font is really 'm' but pretending to be 'ebuc'.

Now if we now ask for 'ub' and the rule above has no 4th argument and the combination ubuc does not exist, then it is tried and we end up with 'ub' (which is then either in existance or is further substituted to 'm' (pretending to be 'ub').

If on the other hand the 4th argument exist we first try 'ebuc' and if that doesn't exist we get the same result as before, but if it does we stay in ebuc.

So I think @user227621 suggestion is ok. 

A possible alternative would be to always fall back to 'b' whenever some sort of bold series is requested as I did for the 'sb' and 'eb' when doing combinations. But I'm not sure this is really that much better. By the end of the day, the basic part of the rules is there to ensure proper changes when a font family has all the fonts that are played with. If not the 4th argument provides a bit of help, but in complicated situations you would need to specify exactly what you want rather than just relying on substitutions.